### PR TITLE
Fix memory leak in Async.AwaitHandle - closes #131

### LINKED ISSUE #131: Fix memory leak in Async.AwaitWaitHandle
From [Codeplex](https://visualfsharp.codeplex.com/workitem/139) opened by [Pepeiborra](https://www.codeplex.com/site/users/view/Pepeiborra)

The details and a simple fix are available at GitHub: 
https://github.com/fsharp/fsharp/pull/333

### PR DESCRIPTION
**Pull request by forki**

### CODE REVIEW

**Comment by rojepp** [comment]
There were tests added to the most recent [PR](https://visualfsharp.codeplex.com/SourceControl/network/forks/Pepeiborra/FsharpAwaitWaitHandle/contribution/7515), you might want to glance at those.

**Comment by forki** [comment]
done

**Comment by dsyme** [comment]
LGTM

File: src/fsharp/FSharp.Core.Unittests/FSharp.Core/Microsoft.FSharp.Control/AsyncModule.fs [hunk 1]
```diff
@@ -191,6 +199,41 @@ type AsyncModule() =
 
         for _i = 1 to 50 do test()
 
+
+    [<Test>]
+    member this.``Async.AwaitWaitHandle does not leak memory`` () =
```
Comment (latkin) [inline src/fsharp/FSharp.Core.Unittests/FSharp.Core/Microsoft.FSharp.Control/AsyncModule.fs]: This test is too expensive. This 1 case by itself takes around 5 min to run, which is somewhere around 70%+ of the overall suite execution time. :hourglass: 

In F# 3.0 I get at least 3 or 4 failures here if I do just 100 iterations.  Reducing to that level cuts the time down to something reasonable.

**Pull request closed without merging**